### PR TITLE
This closes #132 - invalid IDs in URL returns error 500.

### LINKED ISSUE #132: Cast to ObjectId failed for value XXX at path "_id"
From the version/example to date, sending a request with an invalid numeric ID cases the response to be returned status_code 500 with the output:

```
GET /note/1 500 53.133 ms - 131
```

```
{
message: "Cast to ObjectId failed for value "1" at path "_id"",
name: "CastError",
kind: "ObjectId",
value: "1",
path: "_id"
}
```

This happens in case the mongoose schema was designed using default ID values (assumes it as ObjectID instead), causes it to return error 500, which should be 404 (even if you don't check the DB).

More info on this:
[http://stackoverflow.com/questions/14940660/whats-mongoose-error-cast-to-objectid-failed-for-value-xxx-at-path-id](http://stackoverflow.com/questions/14940660/whats-mongoose-error-cast-to-objectid-failed-for-value-xxx-at-path-id)
[http://stackoverflow.com/questions/14940660/whats-mongoose-error-cast-to-objectid-failed-for-value-xxx-at-path-id](http://stackoverflow.com/questions/14940660/whats-mongoose-error-cast-to-objectid-failed-for-value-xxx-at-path-id)

### PR DESCRIPTION
**Pull request by brunobraga**
This fixes the issue.

### CODE REVIEW

**Comment by brunobraga** [comment]
Done, can you please confirm?

Thanks,

File: lib/handlers.js [hunk 1]
```diff
@@ -33,6 +34,10 @@ exports.schema = function(req, res, next) {
 };
 
 exports.get = function(req, res, next) {
+  if (req.params.id && !mongoose.Types.ObjectId.isValid(req.params.id)) {
+    exports.respond(res, 404, exports.objectNotFound());
+    next();
+  } else {
   req.quer.exec(function(err, list) {
```
Comment (baugarten) [inline lib/handlers.js]: Sorry, I hate being nitpicky like this. Would you mind indenting the code inside of the `else` block?

Comment (baugarten) [inline lib/handlers.js]: Sorry, I hate being nitpicky like this. Would you mind indenting the code inside of the else block?

**Comment by brunobraga** [comment]
Done

**Comment by baugarten** [comment]
Thanks!

**Pull request merged**